app/safe/internal/state/stats/stats.go

```diff
@@ -35,36 +40,36 @@ var currentState = Status{
 
 // Increment is a method for the Status struct that increments the NumSecrets
 // field by 1 if the provided secret name is not found in the in-memory store.
-func (s *Status) Increment(name string) {
+func (s *Status) Increment(name string, loader func(name any) (any, bool)) {
```

Comment:
Also, Passing `loader` as an argument is not the best way to do this.

That’s also to fix circular dependency. — Will clean it up in a more targeted PR.